### PR TITLE
Add preliminary environment management

### PR DESCRIPTION
**Pull request by BrianBland**
Add a Create Environment prompt in the environments dropdown.
Add an "X" delete icon to each environment in the same dropdown.
Disallow deleting explicitly-defined ("protected") environments in the API.

TODO:
- [x] Don't display the "remove environment" button for explicitly-defined environments
- [x] Modals instead of prompts for create/remove environment

### CODE REVIEW

**Comment by volkangurel** [comment]
:+1:

**Comment by BrianBland** [comment]
image

File: environments/environments.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+	for _, env := range environments {
+		ret = append(ret, env)
+	}
+	rwMutex.RUnlock()
+	return
+}
+
+// Create creates a new environment with `name`
+func Create(name string) error {
+	_, status, err := kube.Namespaces.Create(&v1.Namespace{
```
Comment (volkangurel) [inline environments/environments.go]: Does kubernetes reject invalid names gracefully?

**Comment by BrianBland** [comment]
image

File: environments/environments.go (hunk 1 above)
Comment (BrianBland) [inline environments/environments.go]: Yeah it won't accept invalid or duplicate namespaces, returning an `*unversioned.Status`

**Pull request merged**